Marlin/Marlin_main.cpp

```diff
@@ -284,11 +284,11 @@ bool axis_homed[3] = { false };
 
 static long gcode_N, gcode_LastN, Stopped_gcode_LastN = 0;
 
-static char* current_command, *current_command_args;
-static int cmd_queue_index_r = 0;
-static int cmd_queue_index_w = 0;
-static int commands_in_queue = 0;
 static char command_queue[BUFSIZE][MAX_CMD_SIZE];
+static char* current_command, *current_command_args;
+static uint8_t cmd_queue_index_r = 0,
+               cmd_queue_index_w = 0,
+               commands_in_queue = 0;
```

Comment:
Contrast this with what happens if I remove the call to `process_next_command`… Of course it totally optimizes out the second test of `commands_in_queue`…
```asm
    20d8:	80 91 b5 03 	lds	r24, 0x03B5

  #if ENABLED(SDSUPPORT)
    card.checkautostart(false);
  #endif

  if (commands_in_queue) {
    20dc:	88 23       	and	r24, r24
    20de:	59 f0       	breq	.+22     	; 0x20f6 <loop+0x28>
    20e0:	81 50       	subi	r24, 0x01	; 1

    // The queue may be reset by a command handler or by code invoked by idle() within a handler
    if (commands_in_queue) {
      --commands_in_queue;
    20e2:	80 93 b5 03 	sts	0x03B5, r24
    20e6:	80 91 b7 03 	lds	r24, 0x03B7
```